### PR TITLE
Fix tests for `C-unwind` ABI changes

### PR DESCRIPTION
**Pull request by hyd-dev**
cc https://github.com/rust-lang/rust/issues/87778

### CODE REVIEW

**Comment by RalfJung** [comment]
> Blocked on rust-lang/rust#87787 now.

I don't understand, that issue just says "Miri doesn't work"... so what is this blocked on?

File: tests/compile-fail/function_calls/exported_symbol_bad_unwind2.rs [hunk 1]
```diff
@@ -14,6 +16,4 @@ fn main() {
     }
     unsafe { nounwind() }
     //[extern_block]~^ ERROR unwinding past a stack frame that does not allow unwinding
```
Comment (RalfJung) [inline tests/compile-fail/function_calls/exported_symbol_bad_unwind2.rs:18]: With `extern_block`, there is no definition, right? Why does this not show an error about an unknown extern function being called...?

Comment (hyd-dev) [reply tests/compile-fail/function_calls/exported_symbol_bad_unwind2.rs:18]: I think the definition is present even with `extern_block`, it just does not have `#[rustc_allocator_nounwind]`.

Comment (RalfJung) [reply tests/compile-fail/function_calls/exported_symbol_bad_unwind2.rs:18]: Oh, right, it's `cfg_attr` not `cfg`... :)

**Comment by RalfJung** [comment]
Bors [seems to be broken](https://rust-lang.zulipchat.com/#narrow/stream/242791-t-infra/topic/bors.20doesn't.20trigger.20auto.20CI.20on.20libc).

**Pull request merged**